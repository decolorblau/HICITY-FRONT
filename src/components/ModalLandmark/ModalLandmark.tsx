import React from "react";
import { StyleSheet, Text, Image } from "react-native";
import { colors, fontSize } from "../../styles/hicity.styles";
import ILandmark from "../../types/landmarkInterface";

interface IModalLandmarkProps {
  landmark: any;
}

const ModalLandmark = ({ landmark }: IModalLandmarkProps) => {
  return (
    <>
      <Image style={styles.modalImage} source={{ uri: landmark.imageUrl }} />
      <Text style={styles.modaltitle}>{landmark.title}</Text>
      <Text style={styles.modalText}>{landmark.introduction}</Text>
    </>
  );
};

const styles = StyleSheet.create({
  modalImage: {
    height: 150,
    width: 150,
  },
  modaltitle: {
    marginTop: 30,
    color: colors.darkGrey,
    fontSize: fontSize.h3,
    fontWeight: "bold",
  },
  modalText: {
    marginBottom: 15,
    color: colors.grey,
  },
});

export default ModalLandmark;
