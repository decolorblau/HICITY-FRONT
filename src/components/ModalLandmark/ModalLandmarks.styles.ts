import { StyleSheet } from "react-native";
import { colors, fontSize } from "../../styles/hicity.styles";

const styles = StyleSheet.create({
  imageContainer: {
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.5,
    shadowRadius: 4,
    position: "absolute",
    left: 20,
    top: -85,
    height: 150,
    width: 150,
    alignItems: "center",
    justifyContent: "center",
  },
  modalImage: {
    height: "100%",
    width: 150,
    borderRadius: 20,
  },
  allText: {
    flexDirection: "row",
  },
  categoryContainer: {
    backgroundColor: colors.yellow,
    height: 25,
    width: 118,
    borderRadius: 50,
    alignItems: "center",
    justifyContent: "center",
    position: "absolute",
    right: 70,
    top: 20,
  },
  category: {
    color: colors.white,
    fontSize: fontSize.extraText,
  },
  modaltitle: {
    marginTop: 55,
    color: colors.darkGrey,
    fontSize: fontSize.h3,
    fontWeight: "bold",
    marginBottom: 5,
  },
  modalText: {
    marginBottom: 15,
    color: colors.grey,
    width: 235,
    height: 70,
  },
  buttonPlay: {
    width: 70,
    height: 70,
    borderRadius: 90,
    backgroundColor: colors.white,
    position: "absolute",
    right: 0,
    top: -120,
  },
  play: {
    color: colors.yellow,
    fontSize: 80,
    top: 0,
    right: -3,
    position: "absolute",
  },
});

export default styles;
