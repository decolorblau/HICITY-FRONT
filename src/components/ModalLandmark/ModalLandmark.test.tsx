import React from "react";
import { render } from "@testing-library/react-native";
import ModalLandmark from "./ModalLandmark";

describe("Given a Register component", () => {
  describe("When it is rendered", () => {
    test("Then screen should match the snapshot", () => {
      const landmark = {
        id: "543534665265",
        title: "Snapshot Test",
        imageUrl: "una-url-de-una-imagen",
        city: "aqui",
        category: "test",
        introduction: "faadfefae",
        description: "djgfglagagkfgafglaflgajdgfkaglagjfd",
        latitude: 3.45343,
        longitude: 6.35345,
        lastUpdate: 14354643646,
      };
      const screen = render(<ModalLandmark landmark={landmark} />);

      expect(screen).toMatchSnapshot();
    });
  });
});
