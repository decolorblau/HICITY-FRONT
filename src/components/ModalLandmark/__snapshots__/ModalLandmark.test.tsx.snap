// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Register component When it is rendered Then screen should match the snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 150,
        "justifyContent": "center",
        "left": 20,
        "position": "absolute",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 4,
        "top": -85,
        "width": 150,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "una-url-de-una-imagen",
        }
      }
      style={
        Object {
          "borderRadius": 20,
          "height": "100%",
          "width": 150,
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffcc00",
        "borderRadius": 50,
        "height": 25,
        "justifyContent": "center",
        "position": "absolute",
        "right": 70,
        "top": 20,
        "width": 118,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 14,
        }
      }
    >
      TEST
    </Text>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View>
      <View>
        <Text
          style={
            Object {
              "color": "#444444",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 5,
              "marginTop": 55,
            }
          }
        >
          Snapshot Test
        </Text>
        <Text
          style={
            Object {
              "color": "#888888",
              "height": 70,
              "marginBottom": 15,
              "width": 235,
            }
          }
        >
          faadfefae
        </Text>
      </View>
    </View>
  </View>,
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 90,
          "height": 70,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": -120,
          "width": 70,
        }
      }
    >
      <Text />
    </View>
  </View>,
]
`;
